### PR TITLE
Break unit converter down into 5 projects (Lesson 1)

### PR DESCRIPTION
**Pull request by maryx**
Add Coding Project 1, Hello Rectangle.

### CODE REVIEW

File: course/02_category_widget/solution_02_category_widget/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// You can read about packages here: https://flutter.io/using-packages/
+import 'package:flutter/material.dart';
+
+// We can also import files from relative paths
+import 'category.dart';
```
Comment (pq) [inline course/02_category_widget/solution_02_category_widget/lib/main.dart]: I *highly* recommend not suggesting folks mix relative and `package` imports.  That gives our tools fits.  See for example:

https://github.com/dart-lang/sdk/issues/32042

File: course/02_category_widget/task_02_category_widget/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            height: 200.0,
+            width: 200.0,
+            // TODO: Determine what properties we'll need to pass into our widget
+            child: new Category(),
+          ),
+        ),
+      ),
+    );
+  }
+}
```
Comment (pq) [inline course/02_category_widget/task_02_category_widget/lib/main.dart]: Add a trailing newline?

File: course/02_category_widget/task_02_category_widget/lib/category.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    //    - The InkWell will not show any animation if the onTap function is
+    //      null. Just make it print something, i.e.:
+    //      ```
+    //       onTap: () {
+    //         print('I was tapped!');
+    //        },
+    //      ```
+    return new Container();
+  }
+}
```
Comment (pq) [inline course/02_category_widget/task_02_category_widget/lib/category.dart]: Add a trailing newline?

**Review by pq — COMMENTED** [review]
A few nits.  More detailed feedback soon! 👍

File: course/01_hello_rectangle/solution_01_hello_rectangle/lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import 'package:flutter/material.dart';
+
+const _padding = const EdgeInsets.all(16.0);
+
+void main() {
+  runApp(new MaterialApp(
+    title: 'Hello Rectangle App Title',
+    home: new Scaffold(
+      appBar: new AppBar(
+        title: new Text('Hello Rectangle App Bar Title'),
```
Comment (pq) [inline course/01_hello_rectangle/solution_01_hello_rectangle/lib/main.dart]: Consider using the `const` constructor?

File: course/01_hello_rectangle/solution_01_hello_rectangle/lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      child: new Align(
+        alignment: Alignment.topCenter,
+        child: new Container(
+          color: Colors.greenAccent,
+          height: 200.0,
+          width: 300.0,
+          child: new Center(
+            child: new Padding(
+              padding: _padding,
+              child: new Text(
```
Comment (pq) [inline course/01_hello_rectangle/solution_01_hello_rectangle/lib/main.dart]: (And here.)

File: course/02_category_widget/solution_02_category_widget/lib/category.dart [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                        size: 60.0,
+                      )
+                    : null,
+              ),
+            ),
+            new Container(
+              height: 40.0,
+              color: Colors.grey[200],
+              child: new Center(
+                child: new Text(
```
Comment (pq) [inline course/02_category_widget/solution_02_category_widget/lib/category.dart]: Again, consider using the `const` constructor here?

**Review by pq — COMMENTED** [review]
A few nits, mainly centering around opportunities to declare `const` constructors where classes are immutable and a preference to use `const` constructors when available (over `new`).  To be honest, I'm not sure if these nits take us too far into the weeds, so do feel free to ignore!

**Comment by maryx** [comment]
@pq, the more nits the better! Thanks so much for reviewing. I do have a habit of forgetting to use const.

File: course/02_category_widget/solution_02_category_widget/lib/main.dart (hunk 1 above)
Comment (maryx) [reply course/02_category_widget/solution_02_category_widget/lib/main.dart]: Thanks, can you please also review? Thanks! https://github.com/flutter/udacity-course/pull/52

Comment (pq) [reply course/02_category_widget/solution_02_category_widget/lib/main.dart]: Done.  Thanks!

**Comment by maryx** [comment]
I will be closing this PR and re-opening it against flutter/udacity-course. Please put review in that PR from now on. Thanks!

**Comment by maryx** [comment]
See https://github.com/flutter/udacity-course/pull/53

**Pull request closed without merging**